src/platform/MpiServer.c

```diff
@@ -0,0 +1,618 @@
+
+typedef enum HTTP_STATUS
+{
+    HTTP_OK = 200,
+    HTTP_BAD_REQUEST = 400,
+    HTTP_NOT_FOUND = 404,
+    HTTP_INTERNAL_SERVER_ERROR = 500
+} HTTP_STATUS;
+
+static char* HttpReasonAsString(HTTP_STATUS statusCode)
+{
+    char* reason = NULL;
+
+    if (NULL == (reason = (char*)malloc(MAX_REASONSTRING_LENGTH)))
+    {
+        return NULL;
+    }
+
+    switch (statusCode)
+    {
+        case HTTP_OK:
+            strcpy(reason, "OK");
+            break;
+        case HTTP_BAD_REQUEST:
+            strcpy(reason, "Bad Request");
+            break;
+        case HTTP_NOT_FOUND:
+            strcpy(reason, "Not Found");
+            break;
+        case HTTP_INTERNAL_SERVER_ERROR:
+            strcpy(reason, "Internal Server Error");
+            break;
+        default:
+            strcpy(reason, "Unknown");
+    }
+
+    return reason;
+}
+
+static const char* JsonGetString(JSON_Object* object, const char* member)
+{
+    const char* value = NULL;
+    JSON_Value* jsonValue = NULL;
+
+    if (object != NULL && member != NULL)
+    {
+        if (NULL != (jsonValue = json_object_get_value(object, member)))
+        {
+            value = json_value_get_string(jsonValue);
+        }
+    }
+
+    return value;
+}
+
+static int JsonGetNumber(JSON_Object* object, const char* member)
+{
+    int value = -1;
+    JSON_Value* jsonValue = NULL;
+
+    if (object != NULL && member != NULL)
+    {
+        if (NULL != (jsonValue = json_object_get_value(object, member)))
+        {
+            value = (int)json_value_get_number(jsonValue);
+        }
+    }
+
+    return value;
+}
+
+static const char* JsonSerializeToString(JSON_Object* object, const char* member)
+{
+    const char* value = NULL;
+    JSON_Value* jsonValue = NULL;
+
+    if (object != NULL && member != NULL)
+    {
+        if (NULL != (jsonValue = json_object_get_value(object, member)))
+        {
+            value = json_serialize_to_string(jsonValue);
+        }
+    }
+
+    return value;
+}
+
+static HTTP_STATUS MpiOpenRequest(JSON_Object* requestObject, char** response, int* responseSize, MPI_HANDLE(*mpiOpen)(const char*, const unsigned int))
+{
+    HTTP_STATUS status = HTTP_OK;
+    char* uuid = NULL;
+    const char* clientName = NULL;
+    int maxPayloadSizeBytes = 0;
+    int estimatedSize = 0;
+    const char* responseFormat = "\"%s\"";
+
+    if (NULL == requestObject)
+    {
+        OsConfigLogError(GetPlatformLog(), "Invalid null request object");
+        return HTTP_BAD_REQUEST;
+    }
+
+    if (NULL == (clientName = JsonGetString(requestObject, g_clientName)))
+    {
+        OsConfigLogError(GetPlatformLog(), "Failed to parse client name from request body");
+        return HTTP_BAD_REQUEST;
+    }
+
+    if (0 > (maxPayloadSizeBytes = JsonGetNumber(requestObject, g_maxPayloadSizeBytes)))
+    {
+        OsConfigLogError(GetPlatformLog(), "Failed to parse max payload size from request body");
+        return HTTP_BAD_REQUEST;
```

Comment:
This is ok but better is to not exit multiple times from the same function. Use a status or 'if else'